target/linux/realtek/files-5.10/drivers/clk/realtek/clk-rtl83xx.c

```diff
@@ -600,8 +592,12 @@ static struct clk_hw *rtcl_get_clkhw(struct of_phandle_args *clkspec, void *prv)
 
 static int rtcl_ccu_register_clocks(void)
 {
+	struct clk *clk;
 	int clk_idx, ret;
 
+	clk = clk_register_fixed_rate(NULL, "xtal_clk", NULL, 0, RTCL_XTAL_RATE);
+	clk_register_clkdev(clk, "xtal_clk", NULL);
+
```

Comment:
Aside from a simpler devicetree definition, what's the benefit of making the driver provide the fixed rate clock input?

In case of a dual-SoC system and synchronous ethernet support (which these chips do appear to support), it might be possible that the secondary SoC is not driven by an external crystal, but could be driven by some external clock generator that needs to be set up first. Having that in the devicetree would provide a proper clock hierarchy. Whereas hard-coding the clock input frequency here omits that explicit relation between the two.

Unless I am misunderstanding the point of this patch, I don't think omitting the 25MHz clock from the board definition (i.e. DTS) is the "right" thing to do. The clock/crystal is an external component, so it should be defined as such IMHO.